TanoaPeakSimulator/TanoaPeakSimulator/Tanoa.py

```diff
@@ -0,0 +1,128 @@
+    generate_output_files(chromosome)
+
+
+# Gets a psudo-random string of a given amount of characters from the main sequence. This will move to taking
+# the buffers on the sides and saving that as well so when sampled later we don't have to return to the
+# Very large string. That's inefficient.
+# Parmeters: sequence_length: Length of the big sample that you want to take
+# bp_length: the length of a smaller base pair that you will sample from the broader region
+#   this is setup so that the buffer can be dynamic based on the size of the reads you want to match
+#   to the original chromosome are
+# Sequence: The actual character sequence, generally attached to a chromosome like 2L or 2R
+# array_of_segments: array of BasePairSegments (samples) from the original chromosome.
+#   This array is stored within a chromosome as bp_sequence_array and is modifying that object.
+def get_random_base_pair_string(sequence_length, bp_length, sequence, array_of_segments):
+    if len(sequence) > sequence_length + 101:  # Ignore sequences that don't have enough length to work...
+        random_start_location = random.randrange(99, len(sequence) - (sequence_length + 1))
+
+        # gets a random sample from the random start location to the sequence length. This eventually should be
+        # random_start_location + sequence_length and + buffer... This is a simple substring
+        mini_sequence = sequence[random_start_location: (random_start_location + sequence_length)]
+        # makes a BasePairSegment object from the mini_sequence string and associated data
+        temp_segment = BasePairSegment(random_start_location, mini_sequence, sequence_length, bp_length)
+        # stores that in the passed array from chromosome
+        array_of_segments.append(temp_segment)
+
+
+# Gets smaller reads from larger samples and returns them as a list
+# Parameters: list_of_bp_segments: the larger samples, stored as an array in a chromosome object
+# bp_length: the length of one single read. Generally between 50-100bps in length
+# bp_amount: how many of these smaller reads you want to generate per larger sample
+# original_sequence: for now, this is the whole original sequence. This parameter will be replaced
+#   by the chromosome and the chromosome will be looped through here, with it storing the sample with
+#   it's bp_length buffers so that we can randomly sample that string. I think that will improve
+#   performance significantly
+def get_random_base_pair_string_list(list_of_bp_segments, bp_length, bp_amount, original_sequence):
+    temp_bp_list = []
+    for x in list_of_bp_segments: # loops through the list of BasePairSegment objects
+        for y in range(bp_amount):
+            random_start_location = random.randrange(x.overlap_start, x.end)  # gets a random location to get a read
+            # from. These are the smaller bp_length reads.
+            temp_bp_list.append(original_sequence[random_start_location: random_start_location + bp_length])
```

Comment:
You probably want to store the chromosome name, start, and end locations of the fake reads.  This way we can see how much error there is when they are aligned back to the reference genome by bwa based on sequence content.

Reply:
if you look at the box folder I put the output files in there for 2R I think. Though I was only doing 100 enhancer zones with 200 fake reads per zone. The fake reads need to be in FASTA format, but are the enhancer regions themselves formatted okay?

Reply:
Ok, just checked.  Yep, that's fine.  Don't need to change the output
format for the regions then.  Thanks!